src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -376,14 +376,28 @@ def _trash_pool_list(self, pool_name):
                 ioctx.set_namespace(namespace)
                 images = self.rbd_inst.trash_list(ioctx)
                 for trash in images:
-                    trash['pool_name'] = pool_name
-                    trash['namespace'] = namespace
-                    trash['deletion_time'] = "{}Z".format(trash['deletion_time'].isoformat())
-                    trash['deferment_end_time'] = "{}Z".format(
-                        trash['deferment_end_time'].isoformat())
-                    result.append(trash)
+                    result.append(self._get_trashed_img(trash, pool_name, namespace))
```

Comment:
Can you add tests?

Just a suggestion: it would be great if the main part of this new logic could be pushed to
src/pybind/mgr/dashboard/services/rbd.py
to favor thin controllers (and keep the logic in the service/model layer).

Reply:
Hmmm good question... I'm not really sure how I can force an RBD to be partially deleted as part of an API test.

Reply:
For the sake of simplicity a refactoring should be done separately. We want to backport this fix down to Nautilus, so huge code changes will make this more difficult than necessary.

Reply:
I created an issue for the refactoring: https://tracker.ceph.com/issues/50901 And I agree... I'm in favor to do it separately after  this PR has been backported. What do you think @alfonsomthd ?